src/RLinearAlgebra.jl

```diff
@@ -23,6 +23,7 @@ module RLinearAlgebra
 
 using LinearAlgebra, Random, Distributions
 
+import StatsBase: sample
```

Comment:
This function is built on top of Random and greatly extends its functionality. It is used for forming subsets of indices without replacement. While you could replace it with `randcycle!` from the random library when sampling is done with equal probability, when we want to do more advance samplers which have different probabilities for each row, this is the only way to do it as far as I can tell. Unless we want to implement all the algorithms ourselves, but this would be unnecessary

Reply:
1. Is there a conflict with `sample` from `StatsBase` and our `sample` function?
2. Where in the code do you use sampling without replacement that is not sampling with equal probability?
3. If you need to sample from a discrete distribution with replacement, we use `Random` and `Distributions` to do this already. I believe `Random` can allow a sampler without replacement.

Reply:
1. To avoid any conflict in the future we can call the `StatsBase` version as `StatsBase.sample`
2. Not yet need, but very next set of samplers will need it
3. `sample` from `StatsBase` is more efficient.